--- nginx-1.19.6.orig/auto/cc/conf
+++ nginx-1.19.6/auto/cc/conf
@@ -3,7 +3,7 @@
 # Copyright (C) Nginx, Inc.
 
 
-LINK="\$(CC)"
+LINK="c++"
 
 MAIN_LINK=
 MODULE_LINK="-shared"
@@ -174,7 +174,7 @@
     . auto/feature
 
     if [ $ngx_found = yes ]; then
-        MAIN_LINK="-Wl,-E"
+        MAIN_LINK="-lxml2 -llzma -lz -lfuzzy -lyajl_s -lcurl -lnghttp2 -lssl -lcrypto -lzstd -lbrotlidec-static"
     fi
 
