diff --git a/images/nginx/rootfs/build.sh b/images/nginx/rootfs/build.sh
index 167485a46..9b8e4c6d9 100755
--- a/images/nginx/rootfs/build.sh
+++ b/images/nginx/rootfs/build.sh
@@ -15,4 +15,5 @@
 # limitations under the License.
 
+set -o xtrace
 set -o errexit
 set -o nounset
@@ -76,43 +77,4 @@ get_src()
 }
 
-# install required packages to build
-apk add \
-  bash \
-  gcc \
-  clang \
-  libc-dev \
-  make \
-  automake \
-  openssl-dev \
-  pcre-dev \
-  zlib-dev \
-  linux-headers \
-  libxslt-dev \
-  gd-dev \
-  geoip-dev \
-  perl-dev \
-  libedit-dev \
-  mercurial \
-  alpine-sdk \
-  findutils \
-  curl ca-certificates \
-  patch \
-  libaio-dev \
-  openssl \
-  cmake \
-  util-linux \
-  lmdb-tools \
-  wget \
-  curl-dev \
-  libprotobuf \
-  git g++ pkgconf flex bison doxygen yajl-dev lmdb-dev libtool autoconf libxml2 libxml2-dev \
-  python3 \
-  libmaxminddb-dev \
-  bc \
-  unzip \
-  dos2unix \
-  yaml-cpp \
-  coreutils
-
 mkdir -p /etc/nginx
 
@@ -490,5 +452,5 @@ CC_OPT="-g -Og -fPIE -fstack-protector-strong \
   -Wno-cast-function-type"
 
-LD_OPT="-fPIE -fPIC -pie -Wl,-z,relro -Wl,-z,now -L$HUNTER_INSTALL_DIR/lib"
+LD_OPT="-fPIE -fPIC -pie -Wl,-z,relro -Wl,-z,now -L$HUNTER_INSTALL_DIR/lib -Wl,-rpath,$LUAJIT_LIB"
 
 if [[ ${ARCH} != "aarch64" ]]; then
@@ -618,7 +580,4 @@ writeDirs=( \
 );
 
-addgroup -Sg 101 www-data
-adduser -S -D -H -u 101 -h /usr/local/nginx -s /sbin/nologin -G www-data -g www-data www-data
-
 for dir in "${writeDirs[@]}"; do
   mkdir -p ${dir};
@@ -628,5 +587,2 @@ done
 rm -rf /etc/nginx/owasp-modsecurity-crs/.git
 rm -rf /etc/nginx/owasp-modsecurity-crs/util/regression-tests
-
-# remove .a files
-find /usr/local -name "*.a" -print | xargs /bin/rm
