diff --git a/images/nginx/rootfs/build.sh b/images/nginx/rootfs/build.sh
index a03f3416f..a2d9b85c8 100755
--- a/images/nginx/rootfs/build.sh
+++ b/images/nginx/rootfs/build.sh
@@ -15,4 +15,5 @@
 # limitations under the License.
 
+set -o xtrace
 set -o errexit
 set -o nounset
@@ -59,4 +60,19 @@ export BUILD_PATH=/tmp/build
 ARCH=$(uname -m)
 
+apk()
+{
+  echo DISABLED apk >&2
+}
+
+addgroup()
+{
+  echo DISABLED addgroup >&2
+}
+
+adduser()
+{
+  echo DISABLED adduser >&2
+}
+
 get_src()
 {
@@ -490,4 +506,6 @@ WITH_MODULES="--add-module=$BUILD_PATH/ngx_devel_kit-$NDK_VERSION \
   --add-module=$BUILD_PATH/ngx_brotli"
 
+LD_OPT+=" -Wl,-rpath,$LUAJIT_LIB"
+
 ./configure \
   --prefix=/usr/local/nginx \
